src/firmware/app/control/trajectory_planner.h

```diff
@@ -1,303 +1,190 @@
 #pragma once
 
+#include <stdbool.h>
+
+#include "firmware/app/control/trajectory_planner_impl.h"
 #include "firmware/shared/math/polynomial_2d.h"
 #include "firmware/shared/math/vector_2d.h"
 
-#define TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS                                              \
-    9000  // The maximum size of the array containing trajectory elements. Assuming the
-          // longest possible path is 9 meters with 1mm segments
+/*
+ * NOTE: constant period means that the duration of time between each point
+ * on the trajectory is CONSTANT. This is different from a constant parameterization
+ * trajectory which has segments defined at constant polynomial evaluation intervals.
+ * Constant arc-length trajectories (not used here) are trajectories where the distance
+ * travelled in each segment in constant.
+ */
 
 typedef struct FirmwareRobotPathParameters
 {
     // The 2D polynomial representation of the path to be followed
     Polynomial2dOrder3_t path;
+    // The 1D polynomial representation of the orientation to be followed
+    Polynomial1dOrder3_t orientation_profile;
     // The path parameterization value indicating the beginning of the
-    // considered path
+    // considered path and orientation profile
     float t_start;
     // The path parameterization value indicating the end of the considered
-    // path
+    // path and orientation profile
     float t_end;
-    // The number of segments to discretize the trajectory into.
-    //  Must be greater than 2 segments.
+    // The number of elements to discretize the trajectory into.
+    //  Must be greater than 2 elements.
     //  THE NUMBER OF SEGMENTS MUST BE UNDER
     //  TRAJECTORY_PLANNER_MAX_NUMBER_SEGMENTS
-    unsigned int num_segments;
+    unsigned int num_elements;
     // The maximum acceleration allowed at any
-    //  point along the trajectory. This factor limits the maximum delta-velocity
-    // and
-    //  also the max speed around curves due to centripetal acceleration [m/s^2]
-    float max_allowable_acceleration;
-    // The maximum speed allowable at any point along the trajectory```
-    float max_allowable_speed;
+    // point along the trajectory. This factor limits the maximum delta-velocity
+    // and also the max speed around curves due to centripetal acceleration [m/s^2]
+    float max_allowable_linear_acceleration;
+    // The maximum allowable linear speed allowable at any point on the
```

Comment:
This is a difficult one because the units are all relative. If the path is in mm then the velocity units need to be mm/s, acceleration mm/s^2.

I'll add a comment indicating this, but I can't add specific units.